packages/common/src/fields/File.ts

```diff
@@ -0,0 +1,52 @@
+import Field from './Field'
+
+type AllowedDisks = "public" | "private"
+
+export class File extends Field {
+    public selectedDisk: AllowedDisks = "public"
+
+    public component = 'FileField'
+
+    /**
+     * When a new file is made, 
+     * we'll allow it to be downloaded
+     *
+     */
+    public constructor(name: string, databaseField?: string) {
+        super(name, databaseField)
+
+        this.attributes = {
+            disableDownload: false
+        }
+    }
+
+    /**
+     * Set a disk on the user's filesystem where the underlying file
+     * would be stored
+     *
+     */
+    public disk(disk: AllowedDisks) {
+        this.selectedDisk = disk
+
+        return this
+    }
+
+    /**
+     * Ensure that users cannot download the
+     * created file
+     *
+     */
+    public disableDownload() {
+        this.attributes = {
```

Comment:
Instead of using spread, use this instead: `this.attributes.disableDownload = true`. By default it should be false.